src/lib/lttng-ctl/lttng-ctl.c

```diff
@@ -287,6 +287,49 @@ int lttng_check_tracing_group(void)
 	return ret;
 }
 
+static int check_enough_available_memory(size_t num_bytes_requested_per_cpu)
+{
+	long num_cpu;
+	size_t best_mem_info;
+	size_t num_bytes_requested_total;
+
+	/*
+	 * Get the number of CPU currently online to compute the amount of
+	 * memory needed to create a buffer for every CPU.
+	 */
+	num_cpu = sysconf(_SC_NPROCESSORS_ONLN);
+	if (num_cpu == -1) {
+		goto error;
+	}
+
+	num_bytes_requested_total = num_bytes_requested_per_cpu * num_cpu;
+
+	/*
+	 * Try to get the `MemAvail` field of `/proc/meminfo`. This is the most
+	 * reliable estimate we can get but it is only exposed by the kernel
+	 * since 3.14. (See Linux kernel commit:
+	 * 34e431b0ae398fc54ea69ff85ec700722c9da773)
+	 */
+	best_mem_info = utils_get_memory_available();
+	if (best_mem_info >= 0) {
+		goto success;
+	}
+
+	/*
+	 * As a backup plan, use `MemTotal` field of `/proc/meminfo`. This
+	 * is a sanity check for obvious user error.
+	 */
+	best_mem_info = utils_get_memory_total();
+	if (best_mem_info >= 0) {
```

Comment:
Since `utils_get_memory_total()` returns a `size_t` (unsigned), this will always be true.